crypto/bio/bio_lib.c

```diff
@@ -45,22 +45,21 @@ int BIO_set(BIO *bio, const BIO_METHOD *method)
     bio->references = 1;
     bio->num_read = 0L;
     bio->num_write = 0L;
-    if (!CRYPTO_new_ex_data(CRYPTO_EX_INDEX_BIO, bio, &bio->ex_data))
-        return 0;
-
     bio->lock = CRYPTO_THREAD_lock_new();
```

Comment:
I don't understand the code-meaning your 1st sentence.
Will reading this code, I also think that the initialisation code should be moved to BIO_new().
Let's do it...